### DISCUSSION TITLE
One-stop shop for theming LXQt and apps

### DISCUSSION THREAD

**Discussion by AzumaHazuki** [lxqt/lxqt#2145] [General]
There are a _lot_ of moving parts that go into making a coherent look and feel for LXQt, possibly more than any other DE I've ever seen. Qt theme, GTK2 theme, GTK3 theme, LXQt theme, and then there's theming whatever WM the user happens to be running (Openbox by default, for now...).

Should we maybe revamp the Appearance/Theming section? I was thinking something with a few notebook tabs up top to separate things into categories: Fonts and font effects (scaling, DPI, antialiasing, hinting) in one tab, icons and wallpaper on the next, LXQt theme on the next, Qt settings next, GTK settings after? It would look a little like Xfce's Appearance dialogue, is my vision.

We could also use some kind of selector for Qt theming in there. Kvantum and its theme manager are separate applications, qt5ct conflicts with KDE's theme management, QtCurve and Kvantum are doing the same thing differently, there's the default Fusion theme that comes with Qt itself, there's work on editing Qt themes' palettes that apply to some themes but not all, etc etc etc. I wish there were a way of consolidating and simplifying all this in one place.

A modular approach springs to mind, that is, the Appearance control panel could check for the Kvantum manager, QtCurve, etc on every startup and add options to itself to launch their respective theme managers if detected. This would avoid integrating them too tightly with LXQt, keep our dependencies down, and contribute toward flexibility, sane defaults, and agnosticism.

That said, it might be worth adding qt5ct or reusing parts of it...

**Comment by tsujan** [comment]
Scaling is definitely missing. I wanted to add it but haven't found the time yet. Hopefully it'll be in LXQt 1.1.0.

**Comment by tsujan** [comment]
> That said, it might be worth adding qt5ct or reusing parts of it...

`qt5ct` is incompatible with LXQt (as it is with KDE). LXQt has its own plugins; if you use `qt5ct`, you'll have to disable them.

`qt5ct` is mostly for GTK DEs that lack a good Qt plugin.

**Comment by AzumaHazuki** [comment]
Fair enough. What do you think of the idea about revamping the Appearance settings as above though? Including dynamic detection of kvantum/qtcurve engines and their respective managers?

**Reply by tsujan** [reply]
> We could also use some kind of selector for Qt theming in there

We already have that, in Qt Style drop-down list:

![lxqt_appearance](https://user-images.githubusercontent.com/981076/145148144-fd5aff62-b525-491b-9f66-8d161c2dad56.png)

> Including dynamic detection of kvantum/qtcurve engines and their respective managers?

All theme engines (widget styles) are listed. How a style shows its settings depends on how it's made. Kvantum adds an entry to Configuration Center:

![config_center](https://user-images.githubusercontent.com/981076/145148844-ab2b1a63-d894-4d3e-80b6-4b853c5823a3.png)

I agree that the GUI can be improved but it's already divided into style, icon, font and cursor. More divisions may be confusing to the user.
